health/manager_test.go

```diff
@@ -0,0 +1,109 @@
+package health_test
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/zenoss/zenoss-go-sdk/health"
+	"github.com/zenoss/zenoss-go-sdk/health/target"
+	"github.com/zenoss/zenoss-go-sdk/health/writer"
+	"github.com/zenoss/zenoss-go-sdk/health/writer/mocks"
+)
+
+var _ = Describe("Manager", func() {
+
+	var (
+		err        error
+		ctx        context.Context
+		cancel     context.CancelFunc
+		config     *health.Config
+		targets    []*target.Target
+		dest       *mocks.Destination
+		wr         writer.HealthWriter
+		manager    health.Manager
+		collector  health.Collector
+		msg1, msg2 *target.Message
+	)
+
+	const (
+		testTargetID = "test.target"
+
+		testMetricID   = "test.metric"
+		testCounterID1 = "test.counter1"
+		testCounterID2 = "test.counter2"
+		totalCounterID = "total.counter"
+	)
+
+	Context("RegistrationOnCollect", func() {
+		It("should return nil", func() {
```

Comment:
I don't see a point in this test. We are not verifying anything on manager side we just checking collector errors

Possible solution to test the manager is to create a new Writer implementation. Something like TestWriter that we will use for tests only. It will have the method like getHealth so we will be able to retrieve a data after manager process it